framework/src/actions/MaterialOutputAction.C

```diff
@@ -491,3 +524,45 @@ MaterialOutputAction::materialOutputHelper<RankFourTensor>(const std::string & p
 
   return names;
 }
+
+template <>
+std::vector<std::string>
+MaterialOutputAction::functorMaterialOutputHelper<Real>(const std::string & property_name,
+                                                        const MaterialBase & material,
+                                                        bool get_names_only)
+{
+  std::vector<std::string> names = {property_name + "_out"};
+  if (!get_names_only)
+  {
+    // add a '_out' suffix to avoid name conflicts between the functor property and aux-variable
+    auto params = getParams("FunctorAux", property_name, property_name + "_out", material);
+    _problem->addAuxKernel("FunctorAux", material.name() + property_name, params);
+  }
+
+  return names;
+}
+
+template <>
+std::vector<std::string>
+MaterialOutputAction::functorMaterialOutputHelper<RealVectorValue>(
+    const std::string & property_name, const MaterialBase & material, bool get_names_only)
+{
+  std::array<char, 3> suffix = {{'x', 'y', 'z'}};
+  std::vector<std::string> names(3);
+  for (const auto i : make_range(Moose::dim))
+  {
+    std::ostringstream oss;
+    // add a '_out' suffix to avoid name conflicts between the functor property and aux-variable
+    oss << property_name << "_out_" << suffix[i];
+    names[i] = oss.str();
+
+    if (!get_names_only)
+    {
+      auto params = getParams("FunctorVectorElementalAux", property_name, oss.str(), material);
+      params.set<unsigned int>("component") = i;
+      _problem->addAuxKernel("FunctorVectorElementalAux", material.name() + oss.str(), params);
+    }
+  }
+
+  return names;
```

Comment:
It seems like the only difference between this and the regular material property routine is the aux kernel type name. Can we factor out a helper that takes the aux kernel type name and re-use the code?

Reply:
there s the addition of the out suffix too and the lack of switch between AD and non AD.

I can go either way